src/metabase/api/table.clj

```diff
@@ -44,15 +48,21 @@
    entity_type     (s/maybe TableEntityType)
    visibility_type (s/maybe TableVisibilityType)}
   (api/write-check Table id)
-  (api/check-500 (db/update-non-nil-keys! Table id
-                   :display_name            display_name
-                   :caveats                 caveats
-                   :points_of_interest      points_of_interest
-                   :show_in_getting_started show_in_getting_started
-                   :entity_type             entity_type
-                   :description             description))
-  (api/check-500 (db/update! Table id, :visibility_type visibility_type))
-  (Table id))
+  (let [original-visibility_type (:visibility_type (Table :id id))]
```

Comment:
Since the API and the DB use `snake_case` what to name something like this is a tricky question. But since it's not being used by the API or the DB I would just keep it `lisp-case` and name it `original-visibility-type`. Mixing the two just looks weird :D